bindings/python/src/pgvecto_rs/psycopg/__init__.py

```diff
@@ -0,0 +1,48 @@
+from psycopg import Connection, ProgrammingError
+from psycopg.adapt import Loader, Dumper
+from psycopg.pq import Format
+from psycopg.types import TypeInfo
+from numpy import ndarray
+from ..utils.serializer import from_db_str, to_db_str
+
+__all__ = ["register_vector"]
+
+
+class VectorDumper(Dumper):
+    format = Format.TEXT
+
+    def dump(self, obj):
+        return to_db_str(obj).encode("utf8")
+
+
+class VectorLoader(Loader):
+    format = Format.TEXT
+
+    def load(self, data):
+        if isinstance(data, memoryview):
+            data = bytes(data)
+        return from_db_str(data.decode("utf8"))
+
+
+def register_vector(context: Connection):
+    info = TypeInfo.fetch(context, "vector")
+    register_vector_info(context, info)
+
+
+async def register_vector_async(context: Connection):
+    info = await TypeInfo.fetch(context, "vector")
+    register_vector_info(context, info)
+
+
+def register_vector_info(context: Connection, info: TypeInfo):
+    if info is None:
+        raise ProgrammingError("vector type not found in the database")
+    info.register(context)
+
+    # add oid to anonymous class for set_types
+    text_dumper = type("", (VectorDumper,), {"oid": info.oid})
```

Comment:
Does it have to be an anonymous class?

Reply:
Just a temporary wrapper class, and the value of info.oid is dynamically. Using `type(...)` is enough and clear.

Reply:
When will the `oid` be used?

Why is it an anonymous class instead of a normal one? I don't think a dynamic value will have to make the class anonymous.

Reply:
I tried a normal defined version and it worked.

The oid will be used by the underlaying psycopg things.